### ISSUE TITLE
Or in filters

### ISSUE THREAD

**Issue by jgcb00** [Pentusha/edgeql-qb#87]
There is no example in the doc for a request like that : 
```
select Movie {
  title
} filter
  .title like 'The %'
  or
  .actors.name = 'Scarlett Johansson';
```

Is it implemented ? I can add it to the docs if you give me a code example of that

Source : [Edge DB Tutorial](https://www.edgedb.com/tutorial/basic-queries/config/by-expression)

**Comment by jgcb00** [comment]
Thanks,
Indeed it's very similar. But how would you do a and in the same or, like this for example ? 
```
select Gpu 
        filter .is_using = false
        and 
            (.current_user = 'dato' and .last_ending_time < datetime_current())
            or
            (.current_user = 'human' and .last_seen_usage < datetime_current() - <duration>'01:00:00')
        and .memory_used < 250
  ```

**Comment by Pentusha** [comment]
Thank you for your interest in the project and the good use cases.
Unfortunately, I found two bugs while trying to build your request.

1. Rendering a function call without arguments caused an error. I fixed this in PR #90.
2. Brackets around `or` operators are not placed correctly. I found a way to build your request without breaking all the other tests. To do this, you need to swap the precedence of `or` and `and` operations [here](https://github.com/Pentusha/edgeql-qb/blob/master/edgeql_qb/operators.py#L57-L58). I'm not sure this is the right solution and should be done this way. At the very least, such a solution [contradicts](https://www.edgedb.com/docs/reference/edgeql/lexical) the EdgeDB documentation. Therefore, I will take some time to dig into the issue more thoroughly. 
If you make the corresponding changes, it will be possible to build your query as follows:
```python
def test_issue_87_3() -> None:
    gpu = EdgeDBModel('Gpu')

    query = (
        gpu
        .select(gpu.c.name)
        .where(gpu.c.is_using is False)
        .where(
            (gpu.c.current_user == 'dato')
            & (gpu.c.last_ending_time < func.std.datetime_current())
            |
            (gpu.c.current_user == 'human')
            & (gpu.c.last_seen_usage < func.std.datetime_current() - timedelta(hours=1)),
        )
        .where(gpu.c.memory_used < types.int16(250))
        .build()
    )
    assert query.query == (
        'select Gpu { name } '
        'filter .is_using = <bool>$filter_0 '
        'and (.current_user = <str>$filter_1 and .last_ending_time < datetime_current()) '
        'or (.current_user = <str>$filter_2 and .last_seen_usage < datetime_current() - <duration>$filter_3) '
        'and .memory_used < <int16>$filter_4'
    )
    assert query.context == FrozenDict(
        filter_0=False,
        filter_1='dato',
        filter_2='human',
        filter_3=timedelta(seconds=3600),
        filter_4=250,
    )
```

**Comment by Pentusha** [comment]
@jgcb00 I'm dig into second case a bit more.

```
filter 
.is_using = false
and (.current_user = 'dato' and .last_ending_time < datetime_current()) or (.current_user = 'human' and .last_seen_usage < datetime_current() - <duration>'01:00:00')
        and .memory_used < 250
```
seems like your predicates could be written in more simple and abstract way like this:
```
filter A and (B and C) or (D and E) and F
where
A := .is_using = false
B := .current_user = 'dato'
C := .last_ending_time < datetime_current()
D := .current_user = 'human'
E := .last_seen_usage < datetime_current() - <duration>'01:00:00'
F := .memory_used < 250
```

We can verify that expressions with parentheses and without parentheses are equivalent as follows:
```python
import itertools


def evaluate_expressions(a, b, c, d, e, f):
    expression1 = a and b and c or d and e and f
    expression2 = a and (b and c) or (d and e) and f
    return expression1, expression2


for a, b, c, d, e, f in itertools.product([False, True], repeat=6):
    expr1, expr2 = evaluate_expressions(a, b, c, d, e, f)
    if expr1 != expr2:
        print(expr1, expr2)
        break
else:
    print('ok')
```
I haven't checked in EdgeQL, but apparently the same logic for placing parentheses applies there too.
So, I think that the fact that parentheses are not being displayed in your expression is more like a feature than a bug.
